source/common/lua/wrappers.cc

```diff
@@ -0,0 +1,27 @@
+#include "common/lua/wrappers.h"
+
+namespace Envoy {
+namespace Lua {
+
+int BufferWrapper::luaByteSize(lua_State* state) {
+  lua_pushnumber(state, data_.length());
+  return 1;
+}
+
+int BufferWrapper::luaGetBytes(lua_State* state) {
+  int index = luaL_checkint(state, 2);
+  int length = luaL_checkint(state, 3);
+  if (index < 0 || length < 0 ||
+      static_cast<uint64_t>(index) + static_cast<uint64_t>(length) > data_.length()) {
+    luaL_error(state, "index/length must be >= 0 and (index + length) must be <= buffer size");
+  }
+
+  // TODO(mattklein123): Reduce copies here by using Lua direct buffer builds.
```

Comment:
What's the tradeoff with using `linearize()` here?

Reply:
linearize() would still require us to copy out to Lua (basically we must always copy to Lua, there is no way to direct access a C buffer), and would also require us to reorganize the buffer. I don't think there would be much benefit from this. Lua does provide a better API for building buffers, but it's a little complicated and I didn't want to deal with it in this PR.